File: DSA_practice/leetcode/palindromeNumber.js
// Given an integer x, return true if x is a palindrome, and false otherwise.

 

// Example 1:

// Input: x = 121
// Output: true
// Explanation: 121 reads as 121 from left to right and from right to left.

var isPalindrome = function(x) {
  if (x<0) return false
    let rem = 0
    let copy = x
  while(x>0){
    rem = rem * 10 + (x%10)
    x = Math.floor(x/10)
  } if(rem===copy){
    return true
  } else return false
};
console.log(isPalindrome(121))

File: python/palindromeNumber.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x<0: return False
        copy = x
        res = 0
        while(x>0):
            res = res * 10 + x%10
            x = x // 10
        if copy != res:
            return False
        else: return True
        

# n = input()
# if n == n[::-1]:
#     print("Palindrome")
# else:
#     print("Not Palindrome")

n = int(input())
if n<0:
    print(False)
else:
    copy = n
    res = 0
    while(n>0):
        res = res * 10 + n%10
        n = n//10
    if copy != res:
      print(False)
    else:
      print(True)